### PR TITLE
ref(envelope): Remove "name" from envelope item headers

### PR DESCRIPTION
**Pull request by jan-auer**
Removes the `name` Item header as per updated spec. Relay should not have to rely on that, and instead infer an appropriate attachment type when parsing multipart requests.

`MultipartItems` now has a way to override the attachment type based on the field name.

cc @Swatinem since this was aligned with the Native SDK

### CODE REVIEW

**Comment by jan-auer** [comment]
Note that this does **not** break backwards compatibility with `sentry-native 0.2.x`

File: relay-server/src/endpoints/common.rs [hunk 1]
```diff
@@ -295,6 +294,7 @@ pub fn cors(app: ServiceApp) -> CorsBuilder<ServiceState> {
             "authentication",
             "authorization",
             "content-encoding",
+            "transfer-encoding",
```
Comment (jan-auer) [inline relay-server/src/endpoints/common.rs:297]: This change is somewhat unrelated. I noticed that the `transfer-encoding` header was missing from CORS, but it is actually allowed and even required for Crashpad uploads. For consistency, I opted to add it here now.

**Review by Swatinem — COMMENTED** [review]
lgtm! I defer preper review to relay devs

**Review by untitaker — APPROVED** [review]
